justfile

```diff
@@ -53,3 +53,23 @@ create-config SHORTNAME FILENAME:
 	@just _run_ops_bin "create_config" "--shortname {{SHORTNAME}} --state-filename $(realpath {{FILENAME}})"
 
 check-chainlist: (_run_ops_bin 'check_chainlist')
+
+check-for-codegen-changes:
+    #!/usr/bin/env bash
+    set -euo pipefail
+    if ! git diff-index --quiet HEAD --; then
+        echo -e "\033[31mError: Working directory is not clean. Please commit or stash your changes before checking for codegen changes.\033[0m"
+        exit 1
+    fi
+    just codegen
```

Comment:
This just generates the addresses JSON based on what’s in the TOML, and doesn’t confirm that either TOML or JSON files are accurate and match the chain state? The codegen check runs on PRs already so I don’t think running this hourly is what we want, unless codegen does check chain state to verify

Reply:
I see, I understood it as [make sure `addresses.json` was up to date](https://github.com/ethereum-optimism/superchain-ops/issues/549). You're right I see this is already being ran in CI.

Reply:
Ah, I think the definition of “up to date” was unclear. CI currently enforces that the JSON is up to date based on the TOML, but the TOML is only updated manually. We should be proactive and check the chain for mismatches between chain data and registry data